src/main/java/com/example/java/maven/game2048/Game.java

```diff
@@ -7,9 +7,74 @@
     public Game(int winningNumber, Board board) {
         this.winningNumber = winningNumber;
         this.board = board;
+        board.placeTwoInitialElementsOnBoard();
+        MessagePrinter.printInstructionsBeforeGame();
     }
 
+
     public void start() {
+        while (!board.checkIfGameOver()) {
+            System.out.println(board);
+            makeValidMove();
+            if (board.checkIfWon(winningNumber)) {
+                break;
+            }
+            board.placeNextElementOnBoard();
+        }
+    }
+
+    private String getPlayerMove() {
+        String userInput = UserInputProvider.getMoveFromPlayer();
+        if (playerMoveIsValid(userInput)) {
+            return userInput;
+        }
+        return getPlayerMove();
+    }
+
+    private boolean playerMoveIsValid(String userInput) {
+        if (userInput.equalsIgnoreCase("a")) {
+            return true;
+        }
+        if (userInput.equalsIgnoreCase("d")) {
+            return true;
+        }
+        if (userInput.equalsIgnoreCase("w")) {
+            return true;
+        }
+        return userInput.equalsIgnoreCase("s");
```

Comment:
minor you can get rid of all ifs and simplify to something like:
```java
return userInput.equalsIgnoreCase("a") || userInput.equalsIgnoreCase("d") || userInput.equalsIgnoreCase("w") || userInput.equalsIgnoreCase("s");
```

Reply:
also I can see that you are using this constructions `userInput.equalsIgnoreCase("expected_input")` in at least two places. So you can introduce methods like:
```java
boolean isMoveLeft(String userInput){
   return userInput.equalsIgnoreCase("a");
}
```
same for other directions (method name can be better :D I am not a master of english XD)

Reply:
Done